numpy/polynomial/_polybase.py

```diff
@@ -479,6 +479,7 @@ def __setstate__(self, dict):
 
     def __call__(self, arg):
         off, scl = pu.mapparms(self.domain, self.window)
+        arg = np.asarray(arg)
```

Comment:
With this, previously working calls using masked arrays would stop working - e.g.,
```
np.polynomial.Polynomial([1, 2])(np.ma.MaskedArray([3, 4], mask=[True, False]))
```

That particular part could be solved by making this `np.asanyarray` - though I guess it is possible the polynomials are also used by, e.g., dask, in which case this would still be a problem.